### PR TITLE
Add stats reporting to FakeAsync

### LINKED ISSUE #248: Feature request: [fake async] add public getters to get the numbers of pending timers
In Angular, we want to assert that no active timers are queued at the end of a fake async tests.

It would be great to add public getters for that:
- `FakeAsync.pendingTimers`
- `FakeAsync.pendingPeriodicTimers`

### PR DESCRIPTION
**Pull request by vicb**
fixes #248

### CODE REVIEW

File: lib/testing/src/async/fake_async.dart [hunk 1]
```diff
@@ -106,6 +106,15 @@ abstract class FakeAsync {
   /// is 1 hour. [timeout] is relative to the elapsed time.
   void flushTimers({Duration timeout: const Duration(hours: 1),
       bool flushPeriodicTimers: true});
+
+  /// The number of created periodic timers that have not been canceled.
+  int periodicTimerCount;
```
Comment (yjbanov) [inline lib/testing/src/async/fake_async.dart]: This defines both a getter and a setter. We only want getter. Make it:

```dart
  int get periodicTimerCount;
```

**Comment by yjbanov** [comment]
Looks great, but let's make the fields read-only.

File: lib/testing/src/async/fake_async.dart [hunk 2]
```diff
@@ -106,6 +106,15 @@ abstract class FakeAsync {
   /// is 1 hour. [timeout] is relative to the elapsed time.
   void flushTimers({Duration timeout: const Duration(hours: 1),
       bool flushPeriodicTimers: true});
+
+  /// The number of created periodic timers that have not been canceled.
+  int get periodicTimerCount;
+
+  /// The number of pending non periodic timers that have not been canceled.
+  int get nonPeriocTimerCount;
```
Comment (yjbanov) [inline lib/testing/src/async/fake_async.dart]: typo: s/Perioc/Periodic/

**Comment by vicb** [comment]
@yjbanov Travis is happy now :)

**Comment by yjbanov** [comment]
LGTM

**Pull request merged**